src/web/handlers/GestureHandler.ts

```diff
@@ -394,62 +397,93 @@ export default abstract class GestureHandler {
+    if (trackerData.size === 0 || !trackerData.has(event.pointerId)) {
+      return;
+    }
 
-      //TODO: Find another way to handle this problem, such as using tracker instead of arrays given in event
-      if (isNaN(id)) return;
+    trackerData.forEach((element: TrackerElement, key: number): void => {
+      const id: number = this.tracker.getMappedTouchEventId(key);
 
       all.push({
         id: id,
-        x: event.allTouches[i].clientX - rect.left,
-        y: event.allTouches[i].clientY - rect.top,
-        absoluteX: event.allTouches[i].clientX,
-        absoluteY: event.allTouches[i].clientY,
+        x: element.lastX - rect.left,
+        y: element.lastY - rect.top,
+        absoluteX: element.lastX,
+        absoluteY: element.lastY,
+      });
+    });
+
+    // Each pointer send its own event, so we want changed touches to contain only the pointer that has changed.
+    // However, if the event is cancel, we want to cancel all pointers to avoid crashes
+    if (event.eventType !== EventTypes.CANCEL) {
+      changed.push({
+        id: this.tracker.getMappedTouchEventId(event.pointerId),
+        x: event.x - rect.left,
+        y: event.y - rect.top,
+        absoluteX: event.x,
+        absoluteY: event.y,
+      });
+    } else {
+      trackerData.forEach((element: TrackerElement, key: number): void => {
+        const id: number = this.tracker.getMappedTouchEventId(key);
+
+        changed.push({
+          id: id,
+          x: element.lastX - rect.left,
+          y: element.lastY - rect.top,
+          absoluteX: element.lastX,
+          absoluteY: element.lastY,
+        });
       });
     }
 
-    // eslint-disable-next-line @typescript-eslint/prefer-for-of
-    for (let i = 0; i < event.changedTouches.length; ++i) {
-      const id: number = this.tracker.getMappedTouchEventId(
-        event.changedTouches[i].identifier
-      );
+    let eventType: TouchEventType = TouchEventType.UNDETERMINED;
 
-      if (isNaN(id)) return;
+    switch (event.eventType) {
+      case EventTypes.DOWN:
+      case EventTypes.ADDITIONAL_POINTER_DOWN:
+        eventType = TouchEventType.DOWN;
+        break;
+      case EventTypes.UP:
+      case EventTypes.ADDITIONAL_POINTER_UP:
+        eventType = TouchEventType.UP;
+        break;
+      case EventTypes.MOVE:
+        eventType = TouchEventType.MOVE;
+        break;
+      case EventTypes.CANCEL:
+        eventType = TouchEventType.CANCELLED;
+        break;
+    }
 
-      changed.push({
-        id: id,
-        x: event.changedTouches[i].clientX - rect.left,
-        y: event.changedTouches[i].clientY - rect.top,
-        absoluteX: event.changedTouches[i].clientX,
-        absoluteY: event.changedTouches[i].clientY,
-      });
+    let numberOfTouches: number = all.length;
+
+    if (
+      event.eventType === EventTypes.UP ||
+      event.eventType === EventTypes.ADDITIONAL_POINTER_UP
+    ) {
+      --numberOfTouches;
```

Comment:
I assume this is to be compatible with native implementation, if so I'd mention it in comment